sov-modules/sov-modules-api/src/bech32.rs

```diff
@@ -0,0 +1,106 @@
+use std::{str::FromStr, fmt::{Display, self}};
+use bech32::{ToBase32, FromBase32, Error};
+use serde::{Deserialize, Deserializer, Serialize, Serializer};
+
+use crate::Address;
+
+pub fn vec_to_bech32(vec: &[u8], hrp: &str) -> Result<String, Error> {        
+    let data = vec.to_base32();
+    let bech32_addr = bech32::encode(hrp, data, bech32::Variant::Bech32)?;
+    Ok(bech32_addr.to_string())
+}
+
+pub fn bech32_to_vec(bech32_addr: &str) -> Result<(String, Vec<u8>), Error> {
+    let (hrp, data, _) = bech32::decode(bech32_addr)?;
+    let vec = Vec::<u8>::from_base32(&data)?;
+    Ok((hrp, vec))
+}
+
+const HRP: &str = "sov";
+
+#[derive(borsh::BorshDeserialize, borsh::BorshSerialize, Debug, PartialEq, Clone, Eq)]
+pub struct AddressBech32{    
+    value: String
+}
+
+impl TryFrom<&[u8]> for AddressBech32 {
+    type Error = bech32::Error;
+
+    fn try_from(addr: &[u8]) -> Result<Self, bech32::Error> {
+        if addr.len() != 32 {
+            return Err(bech32::Error::InvalidLength);
+        }
+        let string = vec_to_bech32(addr, HRP)?;
+        Ok(AddressBech32{ value: string })
+    }    
+}
+
+impl From<&Address> for AddressBech32 {
+    fn from(addr: &Address) -> Self {
+        let string = vec_to_bech32(&addr.addr, HRP).unwrap();
```

Comment:
If this can fail then let's implement `TryFrom ` instead.

Reply:
Yes was thinking about it, but this is guaranteed to be 32 bytes already and prefix is fixed. I think this just can't fail. At least I don't see how it could.

Reply:
ok